File: services/gocore/pkg/utils/cpu_heavy.go
// Package utils
package utils

import "log"

// HeavyCalculation 실제 CPU 연산 로직: 난이도(n)만큼 반복하며 CPU 점유
func HeavyCalculation(n int32) int64 {

	const baseLoop = 100_000
	loopCount := int(n) * baseLoop

	log.Printf("DEBUG: Starting CPU loop. Complexity: %d, Total Loop Count: %d", n, loopCount)

	var sum int64 = 0
	for i := 0; i < loopCount; i++ {
		sum += int64(i)*int64(i)%1_000 + 1
	}

	log.Printf("DEBUG: CPU loop finished.")

	return sum
}

File: services/flask-gateway/utils/cpu_heavy.py
def heavy_calculation(complexity_level: int) -> int:
    """CPU를 점유하는 복잡한 연산을 수행합니다.

    Args:
        complexity_level (int): 연산 난이도

    Returns:
        int: 연산 결과
    """
    BASE_LOOP = 100_000
    loop_count = complexity_level * BASE_LOOP

    sum_val = 0
    for i in range(loop_count):
        sum_val += i * i % 1_000 + 1
    
    return sum_val